lib/src/repo.rs

```diff
@@ -1355,7 +1355,7 @@ impl MutableRepo {
                 .store()
                 .get_commit(&wc_commit_id)
                 .map_err(EditCommitError::WorkingCopyCommitNotFound)?;
-            if wc_commit.is_discardable()?
+            if wc_commit.is_discardable(self)?
                 && self
                     .view
                     .with_ref(|v| local_branch_target_ids(v).all(|id| id != wc_commit.id()))
```

Comment:
Non-blocking: should the lack of branches pointing to the commit be considered part of definition of "discardable"?

Reply:
Seems reasonable, but I think the reason it looks the way it does is so we can access the private `self.view` field without calling `enforce_view_invariants()` as `self.view()` would do.

Reply:
At this point, it could be `repo.is_discardable(commit)`. I think this would sidestep the issue. The point you're making is certainly important, we'd probably want to explain it if we transform the function.

In terms of my motivation: this only matters if `is_discardable` gets another caller, but that's likely if we do something like https://github.com/martinvonz/jj/pull/3777#issuecomment-2136435525.